core/src/main/java/org/apache/druid/data/input/TextReader.java

```diff
@@ -55,7 +56,8 @@ public InputRowSchema getInputRowSchema()
       throws IOException
   {
     final LineIterator delegate = new LineIterator(
-        new InputStreamReader(source.open(), StringUtils.UTF8_STRING)
+        // create a small buffer, otherwise LineIterator creates a default 8KB buffer
+        new BufferedReader(new InputStreamReader(source.open(), StringUtils.UTF8_STRING), 256)
```

Comment:
What is the rational that 256 is used for the buffer size?

Reply:
This is a good question, I chose 256 as it's smaller and captures the majority of the lines we ingest. It seems like something we could make configurable. Thoughts?